modules/javafx.graphics/src/main/java/javafx/css/SimpleSelector.java

```diff
@@ -78,30 +79,65 @@ public String getName() {
      * @return an immutable list of style-classes of the {@code Selector}
      */
     public List<String> getStyleClasses() {
-
-        final List<String> names = new ArrayList<>();
-
-        Iterator<StyleClass> iter = styleClassSet.iterator();
-        while (iter.hasNext()) {
-            names.add(iter.next().getStyleClassName());
-        }
-
-        return Collections.unmodifiableList(names);
+        return List.copyOf(selectorStyleClassNames);
     }
 
     /**
      * Gets the immutable {@code Set} of {@code StyleClass}es of the {@code Selector}.
      * @return the {@code Set} of {@code StyleClass}es
      */
     public Set<StyleClass> getStyleClassSet() {
-        return styleClassSet;
+        if (cachedStyleClasses == null) {
+            cachedStyleClasses = getStyleClassNames().stream().map(SimpleSelector::getStyleClass).collect(Collectors.toUnmodifiableSet());
+        }
+
+        return cachedStyleClasses;
+    }
+
+    /*
+     * Copied from removed StyleClassSet to give StyleClasses a fixed index when
+     * first encountered. No longer needed once StyleClass is removed.
```

Comment:
Is this a possible idea for the future?

Reply:
Yes, once this class is moved to private API (see https://github.com/openjdk/jfx/pull/1333) then this method and `StyleClass` can be removed altogether.  The `SimpleSelector` class is already deprecated for removal since 23.

The method is almost impossible to reach (you'd have to cast to `SimpleSelector`), but removing it as part of this PR would have required a CSR making this harder to integrate.